AIAgent/run_training.py

```diff
@@ -70,72 +69,104 @@ class TrialSettings:
 
 
 def run_training(
-    platforms_config: list[Platform],
     optuna_config: OptunaConfig,
     training_config: TrainingConfig,
     validation_config: ValidationConfig,
     weights_uri: Optional[str],
 ):
-    sampler = optuna.samplers.TPESampler(
-        n_startup_trials=optuna_config.n_startup_trials
-    )
+    def criterion_init():
+        return nn.KLDivLoss(reduction="batchmean")
+
+    if isinstance(validation_config.validation, ValidationWithLoss):
+
+        def validate(model, dataset):
+            return validate_loss(
+                model,
+                dataset,
+                criterion_init(),
+                validation_config.validation.batch_size,
+            )
+    elif isinstance(validation_config.validation, ValidationWithSVMs):
+
+        def load_maps_info():
+            maps: list[GameMap2SVM] = list()
+            for platform in validation_config.validation.platforms_config:
+                svms_info = platform.svms_info
+                for svm_info in svms_info:
+                    for dataset_config in platform.dataset_configs:
+                        dataset_base_path = dataset_config.dataset_base_path
```

Comment:
The body can be moved to top-level function, after that residual code can be inlined as list generator.